numbat/src/parser.rs

```diff
@@ -2,29 +2,56 @@
 //!
 //! Grammar:
 //! ```txt
-//! statement       →   expression | variable_decl | function_decl | dimension_decl | unit_decl | procedure_call | module_import
+//! statement       →   variable_decl | function_decl | dimension_decl | decorator | unit_decl | module_import | procedure_call | expression
+//!
+//! variable_decl   →   "let" identifier ( ":" type_annotation ) ? "=" expression
+//! type_annotation →   boolean | string | dimension_expr
+//! function_decl   →   "fn" identifier ( fn_decl_generic ) ? fn_decl_param "->" type_annotation "=" expression
+//! fn_decl_generic →   "<" ( identifier "," ) * identifier ? ">"
+//! fn_decl_param   →   "(" ( identifier ( ":" dimension_expr ) ? "," )* ( identifier ( ":" dimension_expr ) ) ? ")"
```

Comment:
With #193 close to being merged, we could already change this to
```suggestion
//! fn_decl_param   →   "(" ( identifier ( ":" type_annotation ) ? "," )* ( identifier ( ":" type_annotation ) ) ? ")"
```